test/k8sT/DatapathConfiguration.go

```diff
@@ -104,6 +108,20 @@ var _ = Describe("K8sDatapathConfig", func() {
 			Expect(status.IntOutput()).Should(Equal(3), "Did not find expected number of entries in BPF tunnel map")
 		}
 
+		It("Check connectivity with transparent encryption and VXLAN encapsulation", func() {
+			switch helpers.GetCurrentIntegration() {
+			case helpers.CIIntegrationFlannel:
+				Skip(fmt.Sprintf(
+					"Cilium in %q mode is not supported with transparent encryption and VxLAN. Skipping test.",
+					helpers.CIIntegrationFlannel))
+				return
+			}
+			deployCilium("cilium-ds-patch-vxlan-ipsec.yaml")
+			validateBPFTunnelMap()
+			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test with IPsec between nodes failed")
```

Comment:
Shouldn't we check here that the traffic is correctly encrypted? Any easy way?

Reply:
We should but currently only way would be to get output from netlink. I have an item to add cli support to cilium and then we can check this. I figured I would add the checks here with that PR.